### ISSUE TITLE
Exporter crashes when BGP peer descriptions collector is enabled

### ISSUE THREAD

**Issue by karlism** [tynany/frr_exporter#26]
After an upgrade to 0.2.13 FRR exporter is crashing when BGP peer descriptions collector is enabled:
```
# frr_exporter --collector.bgp.peer-descriptions --collector.bgp.peer-descriptions.plain-text --no-collector.bfd
INFO[0000] Starting frr_exporter (version=0.2.13, branch=HEAD, revision=c83b7d0c4750ad9f0c02e2d85c2d5480361d029d) on :9342  source="frr_exporter.go:118"
panic: inconsistent label cardinality: expected 7 label values but got 6 in []string{"default", "ipv4", "unicast", "64999", "10.XXX.YYY.9", "64999"}

goroutine 20 [running]:
github.com/prometheus/client_golang/prometheus.MustNewConstMetric(...)
	/go/pkg/mod/github.com/prometheus/client_golang@v1.7.1/prometheus/value.go:107
github.com/tynany/frr_exporter/collector.newGauge(0xc0000595c0, 0xc00010bc00, 0x4000000000000000, 0xc0001c0120, 0x6, 0x6)
	/app/collector/collector.go:155 +0xdc
github.com/tynany/frr_exporter/collector.processBGPSummary(0xc0000595c0, 0xc0001b4000, 0x4d2, 0xe00, 0xa24ee2, 0x4, 0xa25da6, 0x7, 0x0, 0xc000041da0)
	/app/collector/bgp.go:366 +0x1460
github.com/tynany/frr_exporter/collector.collectBGP(0xc0000595c0, 0xa24ee2, 0x4)
	/app/collector/bgp.go:297 +0x3f9
github.com/tynany/frr_exporter/collector.(*BGPCollector).Collect(0xedff00, 0xc0000595c0)
	/app/collector/bgp.go:71 +0x40
github.com/tynany/frr_exporter/collector.runCollector(0xc0000595c0, 0xc000012500, 0xc000082a00, 0xc0000278c0)
	/app/collector/collector.go:129 +0xc7
created by github.com/tynany/frr_exporter/collector.(*Exporters).Collect
	/app/collector/collector.go:98 +0x1db
```
It is working properly if I downgrade to 0.2.12.

```
# cat /etc/frr/frr.conf | grep description
 neighbor 10.XXX.YYY.9 description fw1
 neighbor 10.XXX.YYY.10 description fw2
```

**Comment by tynany** [comment]
Thanks for the info. I will take a look at this shortly.

**Comment by tynany** [comment]
I am having a hard time replicating this issue. I have used a testing environment similar to what you described in https://github.com/tynany/frr_exporter/issues/23.

Operating system:
```
# cat /etc/redhat-release
Red Hat Enterprise Linux release 8.3 (Ootpa)
```

FRR version:
```
# yum list installed frr
Installed Packages
frr.x86_64                                                           7.4-01.el8                                                           @frr
```

FRR config:
```
# cat /etc/frr/frr.conf
!
! Ansible managed
!
frr defaults traditional
hostname XXXXXXXXX
log file /var/log/frr/frr.log informational
ip forwarding
no ipv6 forwarding
service integrated-vtysh-config
!
!
! 0.0.0.0
interface eth0
 ip ospf dead-interval 5
 ip ospf hello-interval 1
 ip ospf message-digest-key 1 md5 XXX
!
! 0.0.0.0
interface dummy0
 ip ospf cost 10
!
!
router bgp 64999
 bgp router-id 10.1.1.72
 bgp log-neighbor-changes
 neighbor iBGP peer-group
 neighbor iBGP remote-as 64999
 neighbor 10.1.1.9 peer-group iBGP
 neighbor 10.1.1.9 description fw1
 neighbor 10.1.1.10 peer-group iBGP
 neighbor 10.1.1.10 description fw2
!
router ospf
 ospf router-id 10.3.4.72
 passive-interface default
 no passive-interface eth0
 network 10.3.4.64/26 area 0.0.0.0
 network 10.1.1.128/32 area 0.0.0.0
 area 0.0.0.0 authentication message-digest
!
!
line vty
!
```

frr_exporter version:
```
# ./frr_exporter --version
frr_exporter, version 0.2.13 (branch: HEAD, revision: c83b7d0c4750ad9f0c02e2d85c2d5480361d029d)
  build user:       root@73399596eaad
  build date:       20210411-01:57:17
```

frr_exporter log when scraping (i.e. it doesn't crash):
```
# ./frr_exporter --collector.bgp.peer-descriptions --collector.bgp.peer-descriptions.plain-text --no-collector.bfd 
INFO[0000] Starting frr_exporter (version=0.2.13, branch=HEAD, revision=c83b7d0c4750ad9f0c02e2d85c2d5480361d029d) on :9342  source="frr_exporter.go:118"
```

frr_exporter metric output showing desc:
```
# curl -s localhost:9342/metrics | grep peer_desc
frr_bgp_peer_message_received_total{afi="ipv4",local_as="64999",peer="10.1.1.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_message_received_total{afi="ipv4",local_as="64999",peer="10.1.1.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
frr_bgp_peer_message_sent_total{afi="ipv4",local_as="64999",peer="10.1.1.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_message_sent_total{afi="ipv4",local_as="64999",peer="10.1.1.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
frr_bgp_peer_prefixes_received_count_total{afi="ipv4",local_as="64999",peer="10.1.1.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_prefixes_received_count_total{afi="ipv4",local_as="64999",peer="10.1.1.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
frr_bgp_peer_state{afi="ipv4",local_as="64999",peer="10.1.1.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_state{afi="ipv4",local_as="64999",peer="10.1.1.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
frr_bgp_peer_uptime_seconds{afi="ipv4",local_as="64999",peer="10.1.1.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_uptime_seconds{afi="ipv4",local_as="64999",peer="10.1.1.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
```

Can you please help me fill in the gaps to replicate the issue? Thanks!

**Comment by karlism** [comment]
Thanks for looking into this! I investigated it bit further, tried with various FRR versions (7.5.1 and 7.4) and problem still remains. What I've discovered however is that it works just fine shortly after FRR is restarted but crashes as soon as BGP sessions come up.
While BGP sessions are down:
```
# vtysh -c 'show ip bgp sum'

IPv4 Unicast Summary:
BGP router identifier 10.AAA.BBB.72, local AS number 64999 vrf-id 0
BGP table version 2
RIB entries 3, using 576 bytes of memory
Peers 2, using 43 KiB of memory
Peer groups 1, using 64 bytes of memory

Neighbor        V         AS   MsgRcvd   MsgSent   TblVer  InQ OutQ  Up/Down State/PfxRcd   PfxSnt
10.XXX.YYY.9    4      64999         0         1        0    0    0    never       Active        0
10.XXX.YYY.10   4      64999         0         1        0    0    0    never       Active        0

Total number of neighbors 2
```
```
# curl -s localhost:9342/metrics  | grep peer_desc
frr_bgp_peer_message_received_total{afi="ipv4",local_as="64999",peer="10.XXX.YYY.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_message_received_total{afi="ipv4",local_as="64999",peer="10.XXX.YYY.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
frr_bgp_peer_message_sent_total{afi="ipv4",local_as="64999",peer="10.XXX.YYY.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 1
frr_bgp_peer_message_sent_total{afi="ipv4",local_as="64999",peer="10.XXX.YYY.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 1
frr_bgp_peer_prefixes_received_count_total{afi="ipv4",local_as="64999",peer="10.XXX.YYY.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_prefixes_received_count_total{afi="ipv4",local_as="64999",peer="10.XXX.YYY.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
frr_bgp_peer_state{afi="ipv4",local_as="64999",peer="10.XXX.YYY.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_state{afi="ipv4",local_as="64999",peer="10.XXX.YYY.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
frr_bgp_peer_uptime_seconds{afi="ipv4",local_as="64999",peer="10.XXX.YYY.10",peer_as="64999",peer_desc="fw2",safi="unicast",vrf="default"} 0
frr_bgp_peer_uptime_seconds{afi="ipv4",local_as="64999",peer="10.XXX.YYY.9",peer_as="64999",peer_desc="fw1",safi="unicast",vrf="default"} 0
```
I've also tried enabling `frr_exporter` debug logging but it doesn't produce more info than info logging.

Can you please try to reproduce it with BGP sessions that are up?

**Comment by tynany** [comment]
Thanks for that. I have fixed the bug in commit https://github.com/tynany/frr_exporter/commit/e02fdbb71ba126b77cff9cdde3ae3b5e9f49b7bb. I'll hold off cutting a new version of FRR Exporter until https://github.com/tynany/frr_exporter/pull/27 has been merged.

**Comment by tynany** [comment]
A fix for this issue has been released in [v0.1.14](https://github.com/tynany/frr_exporter/releases/tag/v0.2.14). Please validate and let me know. Thanks!

**Comment by karlism** [comment]
@tynany, just tested `v0.2.14` and it works properly now, thanks for your help!
